documentation for config auto reload feature (#12548)
hashicorp/consul: website/content/docs/agent/options.mdx

  "trace", "debug", "info", "warn", and "err". You can always connect to an agent
  via [`consul monitor`](/commands/monitor) and use any log level. Also,
  the log level can be changed during a config reload.

> - `-auto-reload-config` ((#\_auto_reload_config)) - This flag set Consul to automatically reload
>   [Reloadable Configuration](#reloadable-configuration) when configuration files change.
>   Consul will also watch certificate and key files set in `cert_file` and `key_file` and reload the configuration
>   if updated.

Reviewer: ```suggestion
- `-auto-reload-config` ((#\_auto_reload_config)) - This flag enables Consul to automatically
  [reload configuration files](#reloadable-configuration) when they have changed.
  Consul will also watch certificate and key files specified with the `cert_file` and `key_file` parameters and reload the configuration if updated.
```
Deemphasized "Reloadable Configuration" -- As far as I know, it's not a branded entity. I also tried to be more precise in the second sentence; please ignore if you don't feel that it is an improvement.

- Services
- TLS Configuration
  - Please be aware that this is currently limited to reload a configuration that is already TLS enabled. You cannot enable or disable TLS only with reloading.
  - To avoid a possible security issue, Some parameters part of the TLS Configuration are not automatically reloaded when [-auto-reload-config](#_auto_reload_config) is enabled.
    - [encrypt_verify_incoming](#encrypt_verify_incoming)
    - [verify_incoming](#verify_incoming)
    - [verify_incoming_rpc](#verify_incoming_rpc)
    - [verify_incoming_https](#verify_incoming_https)
    - [verify_outgoing](#verify_outgoing)
    - [verify_server_hostname](#verify_server_hostname)
    - [ca_file](#ca_file)
    - [ca_path](#ca_path)

>     If any of those configurations are changed while [-auto-reload-config](#_auto_reload_config) is enabled,
>     Consul will issue the following warning, `Static Runtime config has changed and need a manual config reload to be applied`.
>     A manual call to `consul reload` or issuing a `SIGHUP` to Consul process would reload the new values.

Reviewer: ```suggestion
    If any of those configurations are changed while [-auto-reload-config](#_auto_reload_config) is enabled,
    Consul will issue the following warning: `Static Runtime config has changed and need a manual config reload to be applied`.
    Make a manual call to `consul reload` or issue a `SIGHUP` to the Consul process to reload the new values.
```
Fixed punctuation and used more direct language to reload the new values.
